### PR TITLE
fix: Avoid exception caused by broadcasting empty result

### LINKED ISSUE #91: Exception occurred while broadcasting an empty result
### Describe the bug

An exception occurs when broadcasting an empty result. I think this is due to the extra bytes being written in the encode compressed stream, causing it to be determined not empty in `CometExec$.decodeBatches`.

error detail:

```
Caused by: java.io.IOException: Unexpected end of input. Missing schema.
	at org.apache.arrow.vector.ipc.ArrowStreamReader.readSchema(ArrowStreamReader.java:205)
	at org.apache.arrow.vector.ipc.ArrowReader.initialize(ArrowReader.java:185)
	at org.apache.arrow.vector.ipc.ArrowReader.ensureInitialized(ArrowReader.java:176)
	at org.apache.arrow.vector.ipc.ArrowReader.getVectorSchemaRoot(ArrowReader.java:73)
	at org.apache.comet.vector.StreamReader.<init>(StreamReader.scala:39)
	at org.apache.spark.sql.comet.execution.shuffle.ArrowReaderIterator.<init>(ArrowReaderIterator.scala:36)
	at org.apache.spark.sql.comet.CometExec$.decodeBatches(operators.scala:135)
	at org.apache.spark.sql.comet.CometBatchRDD.$anonfun$compute$1(CometBroadcastExchangeExec.scala:249)
	at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:293)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at scala.collection.TraversableLike.flatMap(TraversableLike.scala:293)
	at scala.collection.TraversableLike.flatMap$(TraversableLike.scala:290)
	at scala.collection.mutable.ArrayOps$ofRef.flatMap(ArrayOps.scala:198)
	at org.apache.spark.sql.comet.CometBatchRDD.compute(CometBroadcastExchangeExec.scala:249)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:364)
```

### Steps to reproduce

broadcast `tbl_a` in CometBroadcastExchangeExec test case

```
SELECT /*+ BROADCAST(tbl_a) */ tbl_a._1, tbl_b._2 FROM tbl_a JOIN tbl_b " +
                "WHERE tbl_a._1 > tbl_a._2 LIMIT 2
```

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wForget**
## Which issue does this PR close?

Closes #91.

### CODE REVIEW

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala [hunk 1]
```diff
@@ -59,22 +59,28 @@ class CometExecSuite extends CometTestBase {
...
-          }.get.asInstanceOf[CometBroadcastExchangeExec]
-
-          val numParts = nativeBroadcast.executeColumnar().getNumPartitions
-
-          val rows = nativeBroadcast.executeCollect().toSeq.sortBy(row => row.getInt(0))
-          val rowContents = rows.map(row => row.getInt(0))
-          val expected = (0 until numParts).flatMap(_ => (0 until 5).map(i => i + 1)).sorted
-
-          assert(rowContents === expected)
+          Seq(true, false).foreach { emptyBroadcast =>
```
Comment (viirya) [inline spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:62]: Hmm, the test looks complicated by the change. Could you add a new test for empty broadcast case instead of modifying current one?

File: spark/src/main/scala/org/apache/spark/sql/comet/operators.scala [hunk 2]
```diff
@@ -81,7 +81,11 @@ abstract class CometExec extends CometPlan {
 
       out.flush()
       out.close()
-      Iterator((count, cbbos.toChunkedByteBuffer))
+      if (out.size() > 0) {
```
[the remark this answers is not in the document]
Comment (viirya) [reply spark/src/main/scala/org/apache/spark/sql/comet/operators.scala:84]: Oh, I see. The compressed stream is wrapped in `out`. When it is flushed, it might write compression related bytes which are not counted in `out.size()`.

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala (hunk 1 above)
Comment (wForget) [reply spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala]: sure, added

File: spark/src/main/scala/org/apache/spark/sql/comet/operators.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (viirya) [reply spark/src/main/scala/org/apache/spark/sql/comet/operators.scala:84]: I think that is same.

**Review by viirya — APPROVED** [review]

**Review by sunchao — APPROVED** [review]
LGTM too
